### DISCUSSION TITLE
Clone of Page instance?

### DISCUSSION THREAD

**Discussion by indigoviolet** [pymupdf/PyMuPDF#3155] [Q&A] [answered]
I'm wondering if there's a way to make a clone/deepcopy of a Page such that I can draw on it without affecting the original? 

All of the Document methods to add/create new pages are modifying the document itself, but that's not my intent. I just want a Page instance that can I "restore" to after doing some mutations.

**Comment by JorjMcKie** [comment, marked as answer]
You indeed can make a [full copy of the page](https://pymupdf.readthedocs.io/en/latest/document.html#Document.fullcopy_page). But this new page is inside the current document of course.
Although you can of course delete that scratchpad page again.

Another approach would be to make a new, temporary PDF and copy over the page in question. That page copy plus its owning temp PDF are then at your disposal.
